usermanager/src/main/java/org/pesmypetcare/usermanager/datacontainers/user/UserMedalData.java

```diff
@@ -0,0 +1,116 @@
+package org.pesmypetcare.usermanager.datacontainers.user;
+
+
+import androidx.annotation.NonNull;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Oriol Catalán
+ */
+public class UserMedalData {
+    private String name;
+    private List<Double> levels;
+    private String description;
+    private byte[] medalIcon;
+    private Double progress;
+    private Double currentLevel;
+    private List<String> completedLevelsDate;
+
+    /**
+     * Default constructor.
+     */
+    public UserMedalData() {
+    }
+
+    /**
+     * Creates a UserMedalData object with the specified attributes.
+     * @param name UserMedalData's name
+     * @param levels Goals of the UserMedalData's levels
+     * @param description UserMedalData's description
+     * @param medalIcon UserMedalData's icon
+     * @param progress UserMedalData's progress
+     * @param currentLevel UserMedalData's current level of the medal
+     * @param completedLevelsDate UserMedalData's array with the completed levels dates.
+     */
+    public UserMedalData(String name, List<Double> levels, String description,
+                         byte[] medalIcon, Double progress, Double currentLevel,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The user-supplied array 'medalIcon' is stored directly.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5633147)

Reply:
Has de corregir aquest error

Reply:
Has de guardar una copia, ja que java treballa amb referencies